platform/conf/conf.go

```diff
@@ -292,6 +295,78 @@ func (c *Conf) Bytes() []byte {
 	return []byte(c.String())
 }
 
+func (c *Conf) addFileV2(path, filesystem, contents string, mode int) {
+	u, err := url.Parse(dataurl.EncodeBytes([]byte(contents)))
+	if err != nil {
+		return
+	}
+	c.ignitionV2.Storage.Files = append(c.ignitionV2.Storage.Files, v2types.File{
+		Filesystem: filesystem,
+		Path:       v2types.Path(path),
+		Contents: v2types.FileContents{
+			Source: v2types.Url(*u),
+		},
+		Mode: v2types.FileMode(os.FileMode(mode)),
+	})
+}
+
+func (c *Conf) addFileV21(path, filesystem, contents string, mode int) {
+	c.ignitionV21.Storage.Files = append(c.ignitionV21.Storage.Files, v21types.File{
+		Node: v21types.Node{
+			Filesystem: filesystem,
+			Path:       path,
+		},
+		FileEmbedded1: v21types.FileEmbedded1{
+			Contents: v21types.FileContents{
+				Source: dataurl.EncodeBytes([]byte(contents)),
+			},
+			Mode: mode,
```

Comment:
Should this be `&mode`?

Reply:
in `v2.1.0` `mode` is an `int` https://github.com/coreos/mantle/blob/master/vendor/github.com/coreos/ignition/config/v2_1/types/schema.go#L60

Reply:
I figured it was a difference in the spec, but didn't know where to look.  Thanks for the pointer!